core/src/main/java/hudson/PluginManager.java

```diff
@@ -352,8 +354,13 @@ protected PluginManager(ServletContext context, File rootDir) {
         this.context = context;
 
         this.rootDir = rootDir;
-        if(!rootDir.exists())
-            rootDir.mkdirs();
+        if (!Files.isDirectory(rootDir.toPath())) {
```

Comment:
Do we even need to check that?

Reply:
No, the original code was unnecessary. I removed the pointless check.